internal/expr/expr_test.go

```diff
@@ -467,6 +473,10 @@ func (s *ExprSuite) TestPrepareErrors(c *C) {
 		query:       "SELECT * AS &.* FROM t",
 		prepareArgs: []any{struct{ f int }{f: 1}},
 		err:         `cannot prepare expression: cannot use anonymous struct`,
+	}, {
+		query:       "SELECT avg(*) AS &Person.* FROM t",
+		prepareArgs: []any{Person{}},
+		err:         `cannot prepare expression: invalid tag/column name "avg(*)" in "avg(*) AS &Person.*"`,
```

Comment:
Per discussion, let's please constraint it so that both for maps and structs, the field must be named explicitly whenever a function is used, star or not. We can support something later if it beomes clearly necessary, but forbidding funky syntax is the better and safer way to start.